src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -112,6 +112,9 @@ BlockIO InterpreterDropQuery::executeToTable(
             context.checkAccess(table->isView() ? AccessType::DROP_VIEW : AccessType::DROP_TABLE, table_id);
             table->checkTableCanBeDropped();
 
+            if (database->isDictionaryExist(context, table_name))
```

Comment:
It seems this check might be moved into `table->checkTableCanBeDropped()`, i.e. into the function `StorageDictionary::checkTableCanBeDropped()`.

Reply:
Looks better, thanks! Fixed (and added `backQuoteIfNeed`)